tests/text/generate/test_integration_transfomers.py

```diff
@@ -13,21 +13,25 @@ def test_transformers_integration_continuation():
 
     model_name = "hf-internal-testing/tiny-random-GPTJForCausalLM"
     model = models.transformers(model_name, device="cpu")
-    sequence = generate.continuation(model)("Write a short sentence", rng=rng)
+    sequence = generate.continuation(model)("Write a short sentence ", rng=rng)
     assert isinstance(sequence, str)
     assert model.tokenizer.eos_token not in sequence
 
     sequence = generate.continuation(model, max_tokens=10)(
-        "Write a short sentence", rng=rng
+        "Write a short sentence ", rng=rng
     )
     assert isinstance(sequence, str)
 
-    prompts = ["Write a short sentence", "And another one"]
+    prompts = ["Write a short sentence ", "And another one "]
```

Comment:
I am just curious why you added whitespace padding? According to `guidance`, its preferable to terminate prompts without any new space or line, because frequent tokens already come with a space before the word.

Reply:
Good question, that came intuitively with GPT2 and numbers. I'll look at the vocabulary directly to see if that's actually the right thing to do.

Reply:
I came back to this and looked at the vocabulary of the GPT2 tokenizer. It is true that most of the tokens begin with a space.

Your point highlights something that we need to be very careful about, and which might be incorrectly implemented in outlines.

For instance, the regex `[a-z]{3}` will allow "art" to be generated, but not " art". This could make it impossible to generate what would otherwise be the most probable completion. I need to dig more into this. I opened a discussion to keep track of my thinking on this.

Reply:
You are right, in that token healing should be able to correct all these nuances.